src/parquet/column/column-reader-test.cc

```diff
@@ -44,195 +44,135 @@ namespace test {
 
 class TestPrimitiveReader : public ::testing::Test {
  public:
-  void SetUp() {}
-
-  void TearDown() {}
+  void make_pages() {
+    num_values_ = 0;
+    int levels_per_page = num_levels_ / num_pages_;
+    vector<int16_t> defs;
+    vector<int16_t> reps;
+    vector<int32_t> vals;
+    int values_per_page = 0;
+    for (int p = 0; p < num_pages_; p++) {
+      values_per_page = 0;
+      // Create definition levels
+      if (max_def_level_ > 0) {
+        defs.resize(levels_per_page);
+        random_levels(levels_per_page, 0, max_def_level_, defs.data());
+        for (int i = 0; i < levels_per_page; i++) {
+          if (defs[i] == max_def_level_) values_per_page++;
+        }
+        def_levels_.insert(def_levels_.end(), defs.begin(), defs.end());
+      } else {
+        values_per_page = levels_per_page;
+      }
+      // Create repitition levels
+      if (max_rep_level_ > 0) {
+        reps.resize(levels_per_page);
+        random_levels(levels_per_page, 0, max_rep_level_, reps.data());
+        rep_levels_.insert(rep_levels_.end(), reps.begin(), reps.end());
+      }
+      // Create values
+      vals.resize(values_per_page);
+      random_numbers(values_per_page, 0, vals.data());
+      values_.insert(values_.end(), vals.begin(), vals.end());
+      std::shared_ptr<DataPage> page = MakeDataPage<Type::INT32>(vals, defs,
+          max_def_level_, reps, max_rep_level_);
+      pages_.push_back(page);
+
+      num_values_ += values_per_page;
```

Comment:
"Paginating" gets complicated when you have OPTIONAL and REPEATED data. We will have to pass the number of values per page too.